### PR TITLE
Add API endpoints for

### LINKED ISSUE #4: [Post] CRUD API for Post entity
The post entity will have the following endpoints;
- general 
- [ ] list all public post entities by all users(for the frontend)

- user specific
- [ ] list all users post entities
- [ ]  create a post entity
- [ ]  update a post entity
- [ ]  edit a post entity
- [ ] delete a post entity
- [ ]  change post visibility

### PR DESCRIPTION
**Pull request by sonegillis**
## Description
1. Creating, Updating and deleting a land, post and category
2. Uploading images for land and post
3. Getting all land, and post and categories
4. Getting all land, post and categories for a particular user
5. Add post to a list
6. Delete a post from a list

Fixes #4, #5, #8, #12, #13

## How Has This Been Tested?
All the API endpoints in this PR can be seen after running the project locally and visiting _http://localhost:8000/api/doc/swagger/_ or _http://localhost:8000/api/doc/redoc/_

To run the tests for the endpoint do 
```shell
    $ poetry run pytest
```

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added necessary inline code documentation
- [x] I have added tests that prove my fix is effective and that this feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Review by wadleo — CHANGES_REQUESTED** [review]
Hey @sonegillis, can you pull from the develop branch, there are some conflicts here with the latest changes to develop, thanks. Also, when you do that, the circleci build will start passing since it fails because of the old configuration.

**Review by wadleo — CHANGES_REQUESTED** [review]
Also, can you remove all the .idea folder and all the files in it from the commit, also the .python-version file? 
Thanks for adding the media folder to the .gitignore, also can you remove all the files you committed from the media folder.

**Review by gastonche — CHANGES_REQUESTED** [review]
@sonegillis the tests are failing, any idea why?

**Comment by sonegillis** [comment]
> @sonegillis the tests are failing, any idea why?

Hello @gastonche thanks for the review. The codacy review is not really a test but a review for code quality. If you look at the details most of the review checks conflicts (e.g D211 and D203 are conflicting reviews and we can disable one of them). I have also looked at some of the reviews and fixed in the latest commits

**Comment by sonegillis** [comment]
> Also, can you remove all the .idea folder and all the files in it from the commit, also the .python-version file?
> Thanks for adding the media folder to the .gitignore, also can you remove all the files you committed from the media folder.

Hello @wadleo thanks for your review. I have made the changes you requested

**Review by wadleo — CHANGES_REQUESTED** [review]
Hey bro, I see you deleted the .gitignore file and the idea files are still here.

File: .python-version [hunk 1]
```diff
@@ -0,0 +1 @@
+3.7.0
```
Comment (wadleo) [inline .python-version]: @sonegillis please remove this file, thanks.

Comment (wadleo) [reply .python-version]: Also, there are binary files still here after the .python-version file

File: zero_hunger.egg-info/PKG-INFO [hunk 2]
```diff
@@ -0,0 +1,10 @@
+Metadata-Version: 1.2
```
Comment (wadleo) [inline zero_hunger.egg-info/PKG-INFO]: this file also has to be removed

File: zero_hunger.egg-info/SOURCES.txt [hunk 3]
```diff
@@ -0,0 +1,14 @@
+README.md
```
Comment (wadleo) [inline zero_hunger.egg-info/SOURCES.txt]: this too

File: zero_hunger.egg-info/dependency_links.txt [hunk 4]
```diff
@@ -0,0 +1 @@
+
```
Comment (wadleo) [inline zero_hunger.egg-info/dependency_links.txt]: same here

File: zero_hunger.egg-info/entry_points.txt [hunk 5]
```diff
@@ -0,0 +1,3 @@
+[console_scripts]
```
Comment (wadleo) [inline zero_hunger.egg-info/entry_points.txt]: same here

File: zero_hunger.egg-info/requires.txt [hunk 6]
```diff
@@ -0,0 +1,10 @@
+Pillow<8.0.0,>=7.1.2
```
Comment (wadleo) [inline zero_hunger.egg-info/requires.txt]: in short all the whole zero_hunger.egg-info folder should be removed

File: zero_hunger/.env [hunk 7]
```diff
@@ -0,0 +1,18 @@
+DEBUG=True
```
Comment (wadleo) [inline zero_hunger/.env]: Hey bro, please remove this file

**Review by wadleo — CHANGES_REQUESTED** [review]
Removed this files

**Review by wadleo — APPROVED** [review]
Ran it locally, ran tests and LGTM

**Review by gastonche — APPROVED** [review]

**Review by Cressence — APPROVED** [review]

**Pull request merged**